botnim/sync.py

```diff
@@ -223,7 +223,70 @@ def update_assistant(config, config_dir, production, replace_context=False):
         # ...
 
 
-def sync_agents(environment, bots, replace_context=False):
+def delete_common_knowledge_files(client, vector_store_id):
```

Comment:
Generally, the code here should not understand the difference between main and 'common-knowledge' contexts. Context is a list of sources, and do not have any special meaning to this code.
We're not going to keep the 'single vector store' solution as we're moving to elasticsearch, but if we wouldn't this code should have been:
- Get Vector store name  the name of the agent
- For each source, check if we're required to update it:
  - Delete existing files if so
  - Collect file streams for this source (with a different method based on its kind)
- After iterating on all sources, use the `upload file batches` method to update everything.